src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
This could work but is non-trivial. I'd like to push this PR for now as it includes a number of fixes and greatly improves item faults. I'm going to start work on your suggestion in a separate PR